Tangem/Modules/Referral/ReferralView.swift

```diff
@@ -12,63 +12,195 @@ struct ReferralView: View {
     @ObservedObject var viewModel: ReferralViewModel
 
     var body: some View {
-        ScrollView {
-            VStack {
-                Assets.referralDude
-                    .resizable()
-                    .aspectRatio(contentMode: .fit)
-                    .padding(.horizontal, 40)
-                    .frame(maxHeight: 222)
-                
-                Text("referral_title".localized)
-                    .font(Fonts.Bold.title1)
-                    .multilineTextAlignment(.center)
-                    .lineLimit(2)
-                    .padding(.horizontal, 57)
-                    .padding(.top, 20)
-                    .padding(.bottom, 32)
-                
-                content
+        GeometryReader { geometry in
+            ScrollView {
+                VStack(spacing: 0) {
+                    Assets.referralDude
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .padding(.horizontal, 40)
+
+                    Text("referral_title".localized)
+                        .font(Fonts.Bold.title1)
+                        .multilineTextAlignment(.center)
+                        .lineLimit(2)
+                        .padding(.horizontal, 57)
+                        .padding(.top, 28)
+                        .padding(.bottom, 32)
+
+                    content
+                        .padding(.bottom, geometry.safeAreaInsets.bottom == 0 ? 10 : geometry.safeAreaInsets.bottom)
+                    
+                }
+                .frame(minWidth: geometry.size.width, maxWidth: geometry.size.width, minHeight: geometry.size.height + geometry.safeAreaInsets.bottom)
```

Comment:
```suggestion
                .frame(width: geometry.size.width)
                .frame(minHeight: geometry.size.height + geometry.safeAreaInsets.bottom)
```

Смысл в одном и том же значении. А вообще можно еще `.frame(maxWidth: .infinity)`

Reply:
А вообще нужно frame ставить?, `GeometryReader` же и так все пространство займет

Reply:
там проблема с расположением элементов. Они находятся внутри скролла, а кнопка в нижней части экрана должна находиться на дне, да ещё и скролиться. + при разных размерах экрана и если увеличенный шрифт, она может тоже менять местоположение. поэтому приходится считывать размеры и задавать их для контейнера. В UIKit с этим было попроще. Как вариант можно менять весь базовый элемент (скролл) и для загрузчика ставить без скрола, но тогда объемы увеличатся.
Если посмотреть скрины - на девайсах с бровью - места дофига, а на 8+ (на последнем скриншот) уже почти впритык.